src/shims/unix/fs.rs

```diff
@@ -290,22 +289,30 @@ trait EvalContextExtPrivate<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 ("st_gid", 0),
                 ("st_rdev", 0),
                 ("st_atime", access_sec.into()),
-                ("st_atime_nsec", access_nsec.into()),
                 ("st_mtime", modified_sec.into()),
-                ("st_mtime_nsec", modified_nsec.into()),
                 ("st_ctime", 0),
-                ("st_ctime_nsec", 0),
-                ("st_birthtime", created_sec.into()),
-                ("st_birthtime_nsec", created_nsec.into()),
                 ("st_size", metadata.size.into()),
                 ("st_blocks", 0),
                 ("st_blksize", 0),
-                ("st_flags", 0),
-                ("st_gen", 0),
             ],
             &buf,
         )?;
```

Comment:
Seems like at least Illumos has one more field: `st_fstype`. So that should be initialized when running for that OS.

Reply:
unfortunately, libc crate [does not contain it](https://github.com/rust-lang/libc/blob/951cab59826837ccf53e47c75efef53cd3930693/src/unix/solarish/mod.rs#L324). solaris 11.4 and illumos has it tough, but the time it get fixed and release there, it will take a little while.

Reply:
The comment explaining this is still missing.